iml-manager-cli/src/server.rs

```diff
@@ -419,6 +445,86 @@ pub async fn server_cli(command: ServerCommand) -> Result<(), ImlManagerCliError
 
             wait_for_cmds(cmds).await?;
         }
+        ServerCommand::Remove(config) => {
+            let term = Term::stdout();
+
+            let remove_hosts = hostlist_parser::parse(&config.hosts)?;
+
+            tracing::debug!("Parsed hosts {:?}", remove_hosts);
+
+            let api_hosts = get_hosts().await?;
+
+            let unknown_hosts = filter_unknown_hosts(&remove_hosts, &api_hosts.objects);
+
+            for unknown_host in unknown_hosts {
+                display_cancelled(format!(
+                    "Host {} unknown to IML and will not be removed.",
+                    unknown_host.fqdn
+                ));
+            }
+
+            let objects = remove_hosts
+                .iter()
+                .map(|address| TestHostConfig {
+                    address,
+                    auth_type: "existing_keys_choice".into(),
+                })
+                .collect();
+
+            let Objects { objects } = post("test_host", Objects { objects })
+                .await?
+                .error_for_status()?
+                .json()
+                .await
+                .map_err(iml_manager_client::ImlManagerClientError::Reqwest)?;
+
+            let cmds = objects
+                .into_iter()
+                .map(|CmdWrapper { command }| command)
+                .collect();
+
+            term.write_line(&format!("{} preflight checks...", style("Running").green()))?;
+
+            let _cmds = wait_for_cmds(cmds).await?;
+
+            let objects = remove_hosts
+                .iter()
+                .map(|address| HostRemoverConfig {
+                    address: address.to_string(),
+                    verb: "Remove".to_string(),
+                })
+                .collect();
+
+            let Objects { objects }: Objects<CommandAndHostWrapper> =
+                post("/action", Objects { objects })
+                    .await?
+                    .json()
+                    .await
+                    .map_err(iml_manager_client::ImlManagerClientError::Reqwest)?;
+
+            tracing::debug!("command and hosts {:?}", objects);
+
+            let (commands, hosts): (Vec<_>, Vec<_>) = objects
+                .into_iter()
+                .filter_map(|x| x.command_and_host)
+                .map(|x| (x.command, x.host))
+                .unzip();
+
+            term.write_line(&format!("{} servers...", style("Removing").green()))?;
+
+            wait_for_cmds(commands).await?;
+
+            let host_ids: Vec<_> = hosts
+                .iter()
+                .map(|x| ["id__in".into(), x.id.to_string()])
+                .chain(iter::once(["limit".into(), "0".into()]))
+                .collect();
+
+            let ApiList { objects, .. }: ApiList<HostProfileWrapper> =
+                get(HostProfile::endpoint_name(), host_ids).await?;
```

Comment:
I may misunderstand, but if the hosts are removed, will this call make any sense?

Reply:
I misunderstand it too, but It added by point 4 in #1304.